Splunk 6.3.3 Upgrade

jdmurphy opened the pull request:
* Update Splunk version attributes to be specific to Splunk 6.3.3
* Added ChefSpec tests for _install.rb recipe

Tested upgrade path from previous version by provisioning vagrant before and after switching the version attributes.

nmattam commented:
We should also be updating the docs https://github.com/cerner/cerner_splunk/blob/stable/docs/vagrant.md

jdmurphy commented:
@nmattam I updated the docs, rebased and added them to this [commit](https://github.com/cerner/cerner_splunk/pull/109/commits/62642f9875bdd9629fb196adc5a8273323eef93e).

gravesb commented on docs/vagrant.md:
```diff
@@ -11,15 +11,15 @@ Running with Vagrant
     * You can find the suffixes of files at the [Splunk download page](http://splunk.com/download)
-    * Copy a download link, it will look like `http://www.splunk.com/page/download_track?file=6.2.5/splunk/linux/splunk-6.2.5-272645-linux-2.6-x86_64.rpm&platform=...`
+    * Copy a download link, it will look like `http://www.splunk.com/page/download_track?file=6.3.3/splunk/linux/splunk-6.3.3-f44afce176d0-linux-2.6-x86_64.rpm&platform=...`
     * The `file` query parameter will be the suffix you add onto `http://download.splunk.com/releases/`
```
Minor, go ahead and fix this while you're in here. This doc was missed in PR #106. Should be /products instead of /releases.

jdmurphy replied:
Fixed and rebased

gravesb commented on spec/unit/recipes/_install_spec.rb:
```diff
@@ -0,0 +1,215 @@
+      node.set['splunk']['package']['file_suffix'] = '.txt'
+      node.set['splunk']['config']['clusters'] = ['cerner_splunk/cluster']
+    end
+    # Have to include forwarder recipe so that _start recipe can send notifications to services
```
This doesn't make sense here since the forwarder recipe isn't being converged.

jdmurphy replied:
Fixed and rebased

gravesb commented on spec/unit/recipes/_install_spec.rb:
```diff
@@ -0,0 +1,215 @@
+    expect(subject).to include_recipe('cerner_splunk::_restart_marker')
+  end
+
+  it 'does nothing with the splunk-start service and notifies the deletion fo the splunk file marker immediately' do
```
minor, '... deletion _of_ the splunk file ...'.  same below on line 80.

jdmurphy replied:
Fixed and rebased

gravesb commented:
+1

nmattam commented:
The defaults for build and version are also old here. https://github.com/jdmurphy/cerner_splunk/blob/splunk_6_3x/docs/attributes.md#configurable-with-defaults

jdmurphy commented:
@nmattam they were indeed very out of date. I updated and rebased.

acharlieh commented on docs/vagrant.md:
```diff
@@ -11,15 +11,15 @@ Running with Vagrant
+    * Copy a download link, it will look like `http://www.splunk.com/page/download_track?file=6.3.3/splunk/linux/splunk-6.3.3-f44afce176d0-linux-2.6-x86_64.rpm&platform=...`
+    * The `file` query parameter will be the suffix you add onto `http://download.splunk.com/products/splunk/releases/`
+      * In this example the full url is: `https://download.splunk.com/products/splunk/releases/6.3.3/linux/splunk-6.3.3-f44afce176d0-linux-2.6-x86_64.rpm`
+      * You will download the file locally to `<Mirror root>/releases/6.3.3/splunk/linux/splunk-6.3.3-f44afce176d0-linux-2.6-x86_64.rpm
```
Things I should have updated with the switch of URL structures... Mirror will use the same structure as download.splunk.com. `(mirror root)/products/splunk/releases/<version>/...` (assuming we change the mirror root url).

Copy the download link only works for the previous releases anymore... for the most current version you have to inspect the link and you get the full url from the first argument to handleDownload. I wonder if we want to drop these specific instructions to just "Mirror appropriate packages and structure from download.splunk.com

jdmurphy replied:
Since this is for current and future releases I would say we just do your suggestion. They will still be in historical tags if needed.

acharlieh commented:
+1

nmattam commented:
+1
